airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -96,7 +96,9 @@ protected String getCurrentDatabaseName(DSLContext ctx) {
               final String dataType = r.get("data_type", String.class);
               final JsonSchemaPrimitive jsonType = getType(dataType);
               return Field.of(columnName, jsonType);
-            }).collect(Collectors.toList());
+            })
+            .distinct() // this query can return duplicate columns if a column is used in multiple indexes or as a primary key.
+            .collect(Collectors.toList());
```

Comment:
would it make sense to also collect to a set, or does order matter?

Reply:
an interesting question. i think we don't take advantage of it now, but i we may want to have the order in this these lists be the actual order of columns in the table created in the destination.